src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -94,50 +114,66 @@ class MatchService(
 
 fun BoolQueryBuilder.mustWhenPresent(query: String, value: String?): BoolQueryBuilder {
   value.takeIf { !it.isNullOrBlank() }?.let {
-    this.must(QueryBuilders
-        .matchQuery(query, it))
+    this.must(matchQuery(query, it))
+  }
+  return this
+}
+
+fun BoolQueryBuilder.shouldWhenPresent(query: String, value: String?): BoolQueryBuilder {
+  value.takeIf { !it.isNullOrBlank() }?.let {
+    this.should(matchQuery(query, it))
   }
   return this
 }
 
 fun BoolQueryBuilder.mustWhenPresent(query: String, value: LocalDate?): BoolQueryBuilder {
   value?.let {
-    this.must(QueryBuilders
-        .matchQuery(query, it))
+    this.must(matchQuery(query, it))
+  }
+  return this
+}
+
+fun BoolQueryBuilder.shouldWhenPresent(query: String, value: LocalDate?): BoolQueryBuilder {
+  value?.let {
+    this.should(matchQuery(query, it))
   }
   return this
 }
 
 fun BoolQueryBuilder.must(query: String, value: Any): BoolQueryBuilder {
-  this.must(QueryBuilders.matchQuery(query, value))
+  this.must(matchQuery(query, value))
   return this
 }
 
 fun BoolQueryBuilder.mustWhenTrue(predicate: () -> Boolean, query: String, value: String): BoolQueryBuilder {
   value.takeIf { predicate() }?.let {
-    this.must(QueryBuilders
-        .matchQuery(query, it))
+    this.must(matchQuery(query, it))
   }
   return this
 }
 
 fun BoolQueryBuilder.mustMultiMatchKeyword(value: String, vararg query: String): BoolQueryBuilder {
   this.must().add(multiMatchQuery(value, *query)
-      .analyzer("whitespace")
+      .analyzer("keyword")
```

Comment:
decided to use "keyword" which is technically correct, the New Tech code uses "whitespace" which will achieve the same thing, but given the data is parsed as a keyword the ES documentation says we should use the same analyser as the parser